testing/spectest/shared/common/forkchoice/service.go

```diff
@@ -56,3 +63,45 @@ func startChainService(t *testing.T, st state.BeaconState, block block.SignedBea
 	require.NoError(t, service.StartFromSavedState(st))
 	return service
 }
+
+type engineMock struct {
+	powBlocks map[[32]byte]*ethpb.PowBlock
+}
+
+func (m *engineMock) GetPayload(context.Context, [8]byte) (*pb.ExecutionPayload, error) {
+	return nil, nil
+}
+func (m *engineMock) ForkchoiceUpdated(context.Context, *pb.ForkchoiceState, *pb.PayloadAttributes) (*pb.PayloadIDBytes, []byte, error) {
+	return nil, nil, nil
+}
+func (m *engineMock) NewPayload(context.Context, *pb.ExecutionPayload) ([]byte, error) {
+	return nil, nil
+}
+
+func (m *engineMock) LatestExecutionBlock(context.Context) (*pb.ExecutionBlock, error) {
+	return nil, nil
+}
+
+func (m *engineMock) ExchangeTransitionConfiguration(context.Context, *pb.TransitionConfiguration) error {
+	return nil
+}
+
+func (m *engineMock) ExecutionBlockByHash(_ context.Context, hash common.Hash) (*pb.ExecutionBlock, error) {
+	b, ok := m.powBlocks[bytesutil.ToBytes32(hash.Bytes())]
+	if !ok {
+		return nil, nil
+	}
+	tdInBigEndian := bytesutil.ReverseByteOrder(b.TotalDifficulty)
+	tdBigint := new(big.Int)
+	tdBigint.SetBytes(tdInBigEndian)
+	td256, of := uint256.FromBig(tdBigint)
+	if of {
+		return nil, errors.New("could not convert big.Int to uint256")
+	}
+
+	return &pb.ExecutionBlock{
+		ParentHash:      b.ParentHash,
+		TotalDifficulty: td256.String(),
```

Comment:
This combo feels very odd: we get a big endian number that we convert to big int to later convert to uint256 to finally return a string. Surely a bunch of intermediate conversions can be avoided